src/FetchAuthTokenCache.php

```diff
@@ -189,6 +192,15 @@ public function getProjectId(callable $httpHandler = null)
             );
         }
 
+        // Pass the access token from cache for credentials that require an
+        // access token to fetch the project ID. This saves a call to fetch an
+        // access token when a cached token exists.
+        if ($this->fetcher instanceof Credentials\ExternalAccountCredentials) {
```

Comment:
Just double checking since I don't fully understand the caching flow in this library, this call is only used if getProjectId is explicitly called by a user/some other library right? Just want to make sure that the default flow for getting an access token doesn't break if cloudresourcemanager is down.

Reply:
Yes that's correct - this is ONLY for calls to `$credentials->getProjectId()`